### PR TITLE
chore: mantine-fy user completion modal

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #8286 

### Description:

Mantine-fy user completion modal

Remove unused `Checkbox` component

Add new role: `Software Engineer`

screenshots:

Screenshot 2023-12-13 at 10 35 07
Screenshot 2023-12-13 at 10 34 32

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserCompletionModal/index.tsx [hunk 1]
```diff
@@ -63,86 +69,108 @@ const UserCompletionModal: FC = () => {
...
+                centered
+                title={<Title order={4}>Nearly there...</Title>}
+                styles={{
+                    title: {
+                        width: '100%',
+                        textAlign: 'center',
+                    },
+                    header: {
+                        padding: '1rem',
+                        paddingBottom: '0.75rem',
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserCompletionModal/index.tsx:89]: is there a reason why you are using rems and not default sizing properties?

**Review by IrakliJani — COMMENTED** [review]
code looks good, left few questions. testing on render now

**Comment by IrakliJani** [comment]
UI nits

@joaoviana can we make the modal slightly wider?
CleanShot 2023-12-14 at 13 53 08@2x

File: packages/frontend/src/components/UserCompletionModal/index.tsx [hunk 2]
```diff
@@ -1,60 +1,66 @@
...
-
-function shuffleArray(arr: string[]) {
-    return arr
-        .map((value) => ({ value, sort: Math.random() }))
-        .sort((a, b) => a.sort - b.sort)
-        .map(({ value }) => value);
-}
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { shuffle } from 'lodash-es';
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/UserCompletionModal/index.tsx:18]: oops, good point!

File: packages/frontend/src/components/UserCompletionModal/index.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserCompletionModal/index.tsx:89]: great point, let me move the subheading inside the title

**Comment by joaoviana** [comment]
> UI nits
> 
> @joaoviana can we make the modal slightly wider? CleanShot 2023-12-14 at 13 53 08@2x

Screenshot 2023-12-14 at 10 22 30

done!

**Review by IrakliJani — APPROVED** [review]
good stuff!

**Pull request merged**